test/e2e/placement_eviction_test.go

```diff
@@ -29,24 +31,209 @@ var _ = Describe("ClusterResourcePlacement eviction of bound binding, taint clus
 		createWorkResources()
 
 		// Create the CRP.
-		crp := &placementv1beta1.ClusterResourcePlacement{
+		createCRP(crpName)
+	})
+
+	AfterAll(func() {
+		// Remove taint from member cluster 1.
+		removeTaintsFromMemberClusters(taintClusterNames)
+		ensureCRPEvictionDeletion(crpEvictionName)
+		ensureCRPAndRelatedResourcesDeletion(crpName, allMemberClusters)
+	})
+
+	It("should update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), allMemberClusterNames, nil, "0")
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement status as expected")
+	})
+
+	It("should place resources on the all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+	It("add taint to member cluster 1", func() {
+		addTaintsToMemberClusters(taintClusterNames, buildTaints(taintClusterNames))
+	})
+
+	It("create cluster resource placement eviction targeting member cluster 1", func() {
+		crpe := &placementv1alpha1.ClusterResourcePlacementEviction{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpEvictionName,
+			},
+			Spec: placementv1alpha1.PlacementEvictionSpec{
+				PlacementName: crpName,
+				ClusterName:   taintClusterNames[0],
+			},
+		}
+		Expect(hubClient.Create(ctx, crpe)).To(Succeed(), "Failed to create CRP eviction %s", crpe.Name)
+	})
+
+	It("should update cluster resource placement eviction status as expected", func() {
+		crpEvictionStatusUpdatedActual := testutilseviction.StatusUpdatedActual(
+			ctx, hubClient, crpEvictionName,
+			&testutilseviction.IsValidEviction{IsValid: true, Msg: condition.EvictionValidMessage},
+			&testutilseviction.IsExecutedEviction{IsExecuted: true, Msg: condition.EvictionAllowedNoPDBMessage})
+		Eventually(crpEvictionStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement eviction status as expected")
+	})
+
+	It("should ensure no resources exist on evicted member cluster with taint", func() {
+		unSelectedClusters := []*framework.Cluster{memberCluster1EastProd}
+		for _, cluster := range unSelectedClusters {
+			resourceRemovedActual := workNamespaceRemovedFromClusterActual(cluster)
+			Eventually(resourceRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to check if resources doesn't exist on member cluster")
+		}
+	})
+
+	It("should update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), []string{memberCluster2EastCanaryName, memberCluster3WestProdName}, nil, "0")
```

Comment:
nit: it's not very obvious why these two clusters are left while the tainted one is "memberCluster1EastProdName".